Components/FlightControl/Inc/HDITask.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+******************************************************************************
+* File Name          : HDITask.hpp
+* Description        : Primary flight task, default task for the system.
+******************************************************************************
+*/
+#ifndef SOAR_HDI_HPP_
+#define SOAR_HDI_HPP_
+#include "Task.hpp"
+#include "SystemDefines.hpp"
+#include "RocketSM.hpp"
+
+ 
+
+struct BLINK{
+    uint8_t numBlinks;
+    uint16_t delayMs;
+};
+
+
+
+enum STATE_CHANGE {
+    PRELAUNCH = 0,
+    FILL,
+    ARM,
+    IGNITION,
+    LAUNCH,
+    BURN,
+    COAST,
+    DESCENT,
+    RECOVERY,
+    ABORT
+};
```

Comment:
Enum looks like it's not being used. Let's remove it and stay consistent with RocketState in RocketSM.hpp

Reply:
enum STATE_CHANGE is being used in the ```HandleRequestCommand(uint16_t taskCommand)``` in theHDI. cpp file. The other way to have it is the way that RocketSm.cpp has it set up where in line 110 (switch (```rs_currentState->GetStateID())```). However, this way is already implemented when we call on HDI in the RocketSm.cpp in line 77 (```    HDITask::Inst().SendCommand(Command(REQUEST_COMMAND, rs_currentState->GetStateID()));```).

Reply:
RocketSM.cpp:77 ("`HDITask::Inst().SendCommand(Command(REQUEST_COMMAND, rs_currentState->GetStateID()));`") is sending the RocketState enum to HDI. HDI.cpp:119 is interpreting it as a `STATE_CHANGE` enum. To make sure these don't go out of sync, let's just use remove STATE_CHANGE and just use RocketState.